update cmake version being checked to one that works (was: Issue Installing NCEPLIBS-ip)

I'm trying to install this to get to grib2io-interp. I'm on Ubuntu 20.4 and have installed libblas-dev and liblapack-dev. Here are the errors I get:

    jlahowetz2@SleazySloth:/data/temp/forecast-maps/NCEPLIBS-ip-5.1.0$ cmake -DCMAKE_INSTALL_PREFIX=/data/temp/forecast-maps /data/temp/forecast-maps/NCEPLIBS-ip-5.1.0/
    -- The C compiler identification is GNU 9.4.0
    -- The Fortran compiler identification is GNU 9.4.0
    -- Check for working C compiler: /usr/bin/cc
    -- Check for working C compiler: /usr/bin/cc -- works
    -- Detecting C compiler ABI info
    -- Detecting C compiler ABI info - done
    -- Detecting C compile features
    -- Detecting C compile features - done
    -- Check for working Fortran compiler: /usr/bin/gfortran
    -- Check for working Fortran compiler: /usr/bin/gfortran  -- works
    -- Detecting Fortran compiler ABI info
    -- Detecting Fortran compiler ABI info - done
    -- Checking whether /usr/bin/gfortran supports Fortran 90
    -- Checking whether /usr/bin/gfortran supports Fortran 90 -- yes
    -- Library kinds that will be build: 4;d
    -- Setting build type to 'Release' as none was specified.
    -- Looking for Fortran sgemm
    -- Looking for Fortran sgemm - not found
    -- Looking for pthread.h
    -- Looking for pthread.h - found
    -- Performing Test CMAKE_HAVE_LIBC_PTHREAD
    -- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
    -- Looking for pthread_create in pthreads
    -- Looking for pthread_create in pthreads - not found
    -- Looking for pthread_create in pthread
    -- Looking for pthread_create in pthread - found
    -- Found Threads: TRUE  
    -- Looking for Fortran sgemm
    -- Looking for Fortran sgemm - found
    -- Found BLAS: /usr/lib/x86_64-linux-gnu/libblas.so  
    -- Looking for Fortran cheev
    -- Looking for Fortran cheev - not found
    -- Looking for Fortran cheev
    -- Looking for Fortran cheev - found
    -- A library with LAPACK API found.
    failed to create symbolic link '/data/temp/forecast-maps/NCEPLIBS-ip-5.1.0/tests/data' because existing path cannot be removed: Is a directory
    -- Configuring done
    CMake Error at src/CMakeLists.txt:51 (add_library):
      Target "ip_d" links to target "LAPACK::LAPACK" but the target was not
      found.  Perhaps a find_package() call is missing for an IMPORTED target, or
      an ALIAS target is missing?
    
    
    CMake Error at src/CMakeLists.txt:51 (add_library):
      Target "ip_4" links to target "LAPACK::LAPACK" but the target was not
      found.  Perhaps a find_package() call is missing for an IMPORTED target, or
      an ALIAS target is missing?
    
    
    CMake Error at tests/CMakeLists.txt:6 (add_executable):
      Target "test_sptrung_d" links to target "LAPACK::LAPACK" but the target was
      not found.  Perhaps a find_package() call is missing for an IMPORTED
      target, or an ALIAS target is missing?
    Call Stack (most recent call first):
      tests/CMakeLists.txt:159 (create_sp_test)
    
    
    CMake Error at tests/CMakeLists.txt:6 (add_executable):
      Target "test_sptezv_d" links to target "LAPACK::LAPACK" but the target was
      not found.  Perhaps a find_package() call is missing for an IMPORTED
      target, or an ALIAS target is missing?
    Call Stack (most recent call first):
      tests/CMakeLists.txt:157 (create_sp_test)
    
    
    CMake Error at tests/CMakeLists.txt:6 (add_executable):
      Target "test_sptrungv_d" links to target "LAPACK::LAPACK" but the target
      was not found.  Perhaps a find_package() call is missing for an IMPORTED
      target, or an ALIAS target is missing?
    Call Stack (most recent call first):
      tests/CMakeLists.txt:160 (create_sp_test)
    
    
    CMake Error at tests/CMakeLists.txt:6 (add_executable):
      Target "test_splat_d" links to target "LAPACK::LAPACK" but the target was
      not found.  Perhaps a find_package() call is missing for an IMPORTED
      target, or an ALIAS target is missing?
    Call Stack (most recent call first):
      tests/CMakeLists.txt:155 (create_sp_test)
    
    
    CMake Error at tests/CMakeLists.txt:6 (add_executable):
      Target "test_splaplac_d" links to target "LAPACK::LAPACK" but the target
      was not found.  Perhaps a find_package() call is missing for an IMPORTED
      target, or an ALIAS target is missing?
    Call Stack (most recent call first):
      tests/CMakeLists.txt:154 (create_sp_test)
    
    
    CMake Error at tests/CMakeLists.txt:6 (add_executable):
      Target "test_splat_4" links to target "LAPACK::LAPACK" but the target was
      not found.  Perhaps a find_package() call is missing for an IMPORTED
      target, or an ALIAS target is missing?
    Call Stack (most recent call first):
      tests/CMakeLists.txt:155 (create_sp_test)
    
    
    CMake Error at tests/CMakeLists.txt:6 (add_executable):
      Target "test_sptrungv_4" links to target "LAPACK::LAPACK" but the target
      was not found.  Perhaps a find_package() call is missing for an IMPORTED
      target, or an ALIAS target is missing?
    Call Stack (most recent call first):
      tests/CMakeLists.txt:160 (create_sp_test)
    
    
    CMake Error at tests/CMakeLists.txt:49 (add_executable):
      Target "test_earth_radius_d" links to target "LAPACK::LAPACK" but the
      target was not found.  Perhaps a find_package() call is missing for an
      IMPORTED target, or an ALIAS target is missing?
    
    
    CMake Error at tests/CMakeLists.txt:115 (add_library):
      Target "test_library_grib1_4" links to target "LAPACK::LAPACK" but the
      target was not found.  Perhaps a find_package() call is missing for an
      IMPORTED target, or an ALIAS target is missing?
    
    
    CMake Error at tests/CMakeLists.txt:67 (add_executable):
      Target "test_vector_grib2_d" links to target "LAPACK::LAPACK" but the
      target was not found.  Perhaps a find_package() call is missing for an
      IMPORTED target, or an ALIAS target is missing?
    
    
    CMake Error at tests/CMakeLists.txt:67 (add_executable):
      Target "test_vector_grib2_4" links to target "LAPACK::LAPACK" but the
      target was not found.  Perhaps a find_package() call is missing for an
      IMPORTED target, or an ALIAS target is missing?
    
    
    CMake Error at tests/CMakeLists.txt:66 (add_executable):
      Target "test_scalar_grib2_4" links to target "LAPACK::LAPACK" but the
      target was not found.  Perhaps a find_package() call is missing for an
      IMPORTED target, or an ALIAS target is missing?
    
    
    CMake Error at tests/CMakeLists.txt:6 (add_executable):
      Target "test_ncpus_4" links to target "LAPACK::LAPACK" but the target was
      not found.  Perhaps a find_package() call is missing for an IMPORTED
      target, or an ALIAS target is missing?
    Call Stack (most recent call first):
      tests/CMakeLists.txt:153 (create_sp_test)
    
    
    CMake Error at tests/CMakeLists.txt:6 (add_executable):
      Target "test_splaplac_4" links to target "LAPACK::LAPACK" but the target
      was not found.  Perhaps a find_package() call is missing for an IMPORTED
      target, or an ALIAS target is missing?
    Call Stack (most recent call first):
      tests/CMakeLists.txt:154 (create_sp_test)
    
    
    CMake Error at tests/CMakeLists.txt:61 (add_executable):
      Target "tst_gdswzd_grib2_4" links to target "LAPACK::LAPACK" but the target
      was not found.  Perhaps a find_package() call is missing for an IMPORTED
      target, or an ALIAS target is missing?
    
    
    CMake Error at tests/CMakeLists.txt:42 (add_executable):
      Target "test_ipxwafs_4" links to target "LAPACK::LAPACK" but the target was
      not found.  Perhaps a find_package() call is missing for an IMPORTED
      target, or an ALIAS target is missing?
    
    
    CMake Error at tests/CMakeLists.txt:56 (add_library):
      Target "test_library_grib2_4" links to target "LAPACK::LAPACK" but the
      target was not found.  Perhaps a find_package() call is missing for an
      IMPORTED target, or an ALIAS target is missing?
    
    
    CMake Error at tests/CMakeLists.txt:125 (add_executable):
      Target "test_scalar_grib1_4" links to target "LAPACK::LAPACK" but the
      target was not found.  Perhaps a find_package() call is missing for an
      IMPORTED target, or an ALIAS target is missing?
    
    
    CMake Error at tests/CMakeLists.txt:126 (add_executable):
      Target "test_vector_grib1_4" links to target "LAPACK::LAPACK" but the
      target was not found.  Perhaps a find_package() call is missing for an
      IMPORTED target, or an ALIAS target is missing?
    
    
    CMake Error at tests/CMakeLists.txt:66 (add_executable):
      Target "test_scalar_grib2_d" links to target "LAPACK::LAPACK" but the
      target was not found.  Perhaps a find_package() call is missing for an
      IMPORTED target, or an ALIAS target is missing?
    
    
    CMake Error at tests/CMakeLists.txt:49 (add_executable):
      Target "test_earth_radius_4" links to target "LAPACK::LAPACK" but the
      target was not found.  Perhaps a find_package() call is missing for an
      IMPORTED target, or an ALIAS target is missing?
    
    
    CMake Error at tests/CMakeLists.txt:6 (add_executable):
      Target "test_sppad_4" links to target "LAPACK::LAPACK" but the target was
      not found.  Perhaps a find_package() call is missing for an IMPORTED
      target, or an ALIAS target is missing?
    Call Stack (most recent call first):
      tests/CMakeLists.txt:156 (create_sp_test)
    
    
    CMake Error at tests/CMakeLists.txt:6 (add_executable):
      Target "test_fft_4" links to target "LAPACK::LAPACK" but the target was not
      found.  Perhaps a find_package() call is missing for an IMPORTED target, or
      an ALIAS target is missing?
    Call Stack (most recent call first):
      tests/CMakeLists.txt:158 (create_sp_test)
    
    
    CMake Error at tests/CMakeLists.txt:6 (add_executable):
      Target "test_sptrung_4" links to target "LAPACK::LAPACK" but the target was
      not found.  Perhaps a find_package() call is missing for an IMPORTED
      target, or an ALIAS target is missing?
    Call Stack (most recent call first):
      tests/CMakeLists.txt:159 (create_sp_test)
    
    
    CMake Error at tests/CMakeLists.txt:120 (add_executable):
      Target "tst_gdswzd_grib1_d" links to target "LAPACK::LAPACK" but the target
      was not found.  Perhaps a find_package() call is missing for an IMPORTED
      target, or an ALIAS target is missing?
    
    
    CMake Error at tests/CMakeLists.txt:42 (add_executable):
      Target "test_ipxwafs_d" links to target "LAPACK::LAPACK" but the target was
      not found.  Perhaps a find_package() call is missing for an IMPORTED
      target, or an ALIAS target is missing?
    
    
    CMake Error at tests/CMakeLists.txt:120 (add_executable):
      Target "tst_gdswzd_grib1_4" links to target "LAPACK::LAPACK" but the target
      was not found.  Perhaps a find_package() call is missing for an IMPORTED
      target, or an ALIAS target is missing?
    
    
    CMake Error at tests/CMakeLists.txt:56 (add_library):
      Target "test_library_grib2_d" links to target "LAPACK::LAPACK" but the
      target was not found.  Perhaps a find_package() call is missing for an
      IMPORTED target, or an ALIAS target is missing?
    
    
    CMake Error at tests/CMakeLists.txt:125 (add_executable):
      Target "test_scalar_grib1_d" links to target "LAPACK::LAPACK" but the
      target was not found.  Perhaps a find_package() call is missing for an
      IMPORTED target, or an ALIAS target is missing?
    
    
    CMake Error at tests/CMakeLists.txt:6 (add_executable):
      Target "test_sptezv_4" links to target "LAPACK::LAPACK" but the target was
      not found.  Perhaps a find_package() call is missing for an IMPORTED
      target, or an ALIAS target is missing?
    Call Stack (most recent call first):
      tests/CMakeLists.txt:157 (create_sp_test)
    
    
    CMake Error at tests/CMakeLists.txt:61 (add_executable):
      Target "tst_gdswzd_grib2_d" links to target "LAPACK::LAPACK" but the target
      was not found.  Perhaps a find_package() call is missing for an IMPORTED
      target, or an ALIAS target is missing?
    
    
    CMake Error at tests/CMakeLists.txt:6 (add_executable):
      Target "test_sppad_d" links to target "LAPACK::LAPACK" but the target was
      not found.  Perhaps a find_package() call is missing for an IMPORTED
      target, or an ALIAS target is missing?
    Call Stack (most recent call first):
      tests/CMakeLists.txt:156 (create_sp_test)
    
    
    CMake Error at tests/CMakeLists.txt:115 (add_library):
      Target "test_library_grib1_d" links to target "LAPACK::LAPACK" but the
      target was not found.  Perhaps a find_package() call is missing for an
      IMPORTED target, or an ALIAS target is missing?
    
    
    CMake Error at tests/CMakeLists.txt:6 (add_executable):
      Target "test_ncpus_d" links to target "LAPACK::LAPACK" but the target was
      not found.  Perhaps a find_package() call is missing for an IMPORTED
      target, or an ALIAS target is missing?
    Call Stack (most recent call first):
      tests/CMakeLists.txt:153 (create_sp_test)
    
    
    CMake Error at tests/CMakeLists.txt:6 (add_executable):
      Target "test_fft_d" links to target "LAPACK::LAPACK" but the target was not
      found.  Perhaps a find_package() call is missing for an IMPORTED target, or
      an ALIAS target is missing?
    Call Stack (most recent call first):
      tests/CMakeLists.txt:158 (create_sp_test)
    
    
    CMake Error at tests/CMakeLists.txt:126 (add_executable):
      Target "test_vector_grib1_d" links to target "LAPACK::LAPACK" but the
      target was not found.  Perhaps a find_package() call is missing for an
      IMPORTED target, or an ALIAS target is missing?
    
    
    CMake Error at src/CMakeLists.txt:51 (add_library):
      Target "ip_4" links to target "LAPACK::LAPACK" but the target was not
      found.  Perhaps a find_package() call is missing for an IMPORTED target, or
      an ALIAS target is missing?
    
    
    CMake Error at src/CMakeLists.txt:51 (add_library):
      Target "ip_d" links to target "LAPACK::LAPACK" but the target was not
      found.  Perhaps a find_package() call is missing for an IMPORTED target, or
      an ALIAS target is missing?
    
    
    -- Generating done
    CMake Generate step failed.  Build files cannot be regenerated correctly.

--- a/.github/workflows/developer.yml
+++ b/.github/workflows/developer.yml
@@ -42,7 +42,7 @@ jobs:
       run: ctest --test-dir ip/build -j2 --verbose --output-on-failure --rerun-failed
 
     - name: run-gcovr
-      run: gcovr -r ip -v  --html-details  --exclude ip/tests --exclude ip/build/CMakeFiles --print-summary -o test-coverage.html
+      run: gcovr -r ip -v  --html-details  --exclude ip/tests --exclude ip/build/CMakeFiles --gcov-ignore-parse-errors --print-summary -o test-coverage.html
 
     - name: upload-test-coverage
       uses: actions/upload-artifact@v4

--- a/.github/workflows/developer.yml
+++ b/.github/workflows/developer.yml
@@ -26,7 +26,7 @@ jobs:
       run: |
         sudo apt-get update
         sudo apt-get install doxygen libopenblas-dev
-        python3 -m pip install gcovr
+        python3 -m pip install gcovr==7.2
 
     - name: checkout
       uses: actions/checkout@v4
@@ -42,7 +42,7 @@ jobs:
       run: ctest --test-dir ip/build -j2 --verbose --output-on-failure --rerun-failed
 
     - name: run-gcovr
-      run: gcovr -r ip -v  --html-details  --exclude ip/tests --exclude ip/build/CMakeFiles --gcov-ignore-parse-errors --print-summary -o test-coverage.html
+      run: gcovr -r ip -v  --html-details  --exclude ip/tests --exclude ip/build/CMakeFiles --print-summary -o test-coverage.html
 
     - name: upload-test-coverage
       uses: actions/upload-artifact@v4

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,7 +1,7 @@
 # This is the main CMake file for NCEPLIBS-ip.
 #
-# Mark Potts, Kyle Gerheiser, Eric Engle, Ed Hartnett
-cmake_minimum_required(VERSION 3.15)
+# Mark Potts, Kyle Gerheiser, Eric Engle, Ed Hartnett, Alex Richert
+cmake_minimum_required(VERSION 3.18)
 
 # Get the version from the VERSION file.
 file(STRINGS "VERSION" pVersion)

--- a/spack/package.py
+++ b/spack/package.py
@@ -67,6 +67,7 @@ class Ip(CMakePackage):
     depends_on("sp precision=d", when="@4.1:4 precision=d")
     depends_on("sp precision=8", when="@4.1:4 precision=8")
     depends_on("lapack", when="@5.1:")
+    depends_on("cmake@3.18:", when="@5.1:")
 
     def cmake_args(self):
         args = [

